File: Easy/2025-08-05-2107-FruitsIntoBasketsII/solution.java
class Solution {
    public int numOfUnplacedFruits(int[] fruits, int[] baskets) {
        boolean[] used = new boolean[baskets.length];
        int unplaced = 0;
        
        for (int fruit : fruits) {
            boolean placed = false;
            
            // Find the leftmost available basket with sufficient capacity
            for (int i = 0; i < baskets.length; i++) {
                if (!used[i] && baskets[i] >= fruit) {
                    used[i] = true;  // Mark basket as used
                    placed = true;
                    break;
                }
            }
            
            if (!placed) {
                unplaced++;
            }
        }
        
        return unplaced;
    }
} 

File: Easy/2025-08-05-2107-FruitsIntoBasketsII/solution.py
from typing import List

class Solution:
    def numOfUnplacedFruits(self, fruits: List[int], baskets: List[int]) -> int:
        used = [False] * len(baskets)
        unplaced = 0
        
        for fruit in fruits:
            placed = False
            
            # Find the leftmost available basket with sufficient capacity
            for i in range(len(baskets)):
                if not used[i] and baskets[i] >= fruit:
                    used[i] = True  # Mark basket as used
                    placed = True
                    break
            
            if not placed:
                unplaced += 1
        
        return unplaced 